src/main/resources/META-INF/resources/primefaces/datatable/datatable.js

```diff
@@ -4561,6 +4563,32 @@ PrimeFaces.widget.DataTable = PrimeFaces.widget.DeferredWidget.extend({
         this.clone.find('.ui-column-filter').prop('disabled', true);
     },
 
+    /**
+     * Initializes the expansion state
+     * @private
+     */
+    initRowExpansion: function() {
+        var $this = this;
+
+    	this.expansionHolder = $(this.jqId + '_rowExpansionState');
+        this.loadedExpansionRows = this.tbody.children('.ui-expanded-row-content').prev().map(function() {
+            return $this.getRowMeta($(this)).index;
+        }).get();
+
+        this.writeRowExpansions();
+    },
+    
+    writeRowExpansions: function() {
+    	this.expansionHolder.val(this.loadedExpansionRows.join(','));
+    },
+
+    rowExpansionLoaded: function(rowIndex) {
+    	if (!PrimeFaces.inArray(this.loadedExpansionRows, rowIndex)) {
+    		this.loadedExpansionRows.push(rowIndex);
+    		this.writeRowExpansions();
+    	}
+    },
```

Comment:
Looks like your formatting spacing here in JavaScript is a little off compared to rest of the file?